src/structures/Emoji.js

```diff
@@ -124,6 +124,39 @@ class Emoji {
   }
 
   /**
+   * Set the name of the emoji
+   * @param {string} name The new name for the emoji
+   * @returns {Promise<Emoji>}
+   */
+  setName(name) {
+    return this.edit({ name });
+  }
+
+  /**
+   * Add a role to the list of roles that can use this emoji
+   * @param {Role} role The role to add
+   * @returns {Promise<Emoji>}
+   */
+  addRestrictedRole(role) {
```

Comment:
this should probably just call `this.addRestrictedRoles([role])` for consistancy